pkg/pgmodel/model/pgsafetype/custom_pgtypes.go

```diff
@@ -0,0 +1,99 @@
+package pgsafetype
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/jackc/pgtype"
+	"github.com/timescale/promscale/pkg/pgmodel/model"
+)
+
+// Text is a custom pgtype that wraps the pgtype.Text. It is safe from null characters.
+type Text struct {
+	pgtype.Text
+}
+
+func (t *Text) Set(src interface{}) error {
+	text := src.(string)
+	if err := t.Text.Set(sanitizeNullChars(text)); err != nil {
+		return fmt.Errorf("safe-text: %w", err)
+	}
+	return nil
+}
+
+func (t *Text) Get() interface{} {
+	return revertSanitization(t.Text.Get().(string))
+}
+
+func (t *Text) AssignTo(dst interface{}) error {
```

Comment:
don't we need to revertSanitization after assignment?

Reply:
I too felt that we will require here, but it seems that the only place that can be used to handle the reverting stuff is the Get(). Seems like Get() is called every time a value is required, which is why you can see from the end to end tests in `null_chars_test.go`.